designs/2019-additional-lint-targets/README.md

```diff
@@ -0,0 +1,133 @@
+- Start Date: 2019-05-12
+- RFC PR:
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Configuring Additional Lint Targets with `.eslintrc`
+
+## Summary
+
+This proposal adds the ability to specify additional target files into configuration files. This enhancement will solve the pain that people have to use the `--ext` option with wanted file extensions even if they use plugins which support additional file types.
+
+## Motivation
+
+People have to use the `--ext` option or glob patterns to check wanted files even if they use plugins which support additional file types.
+
+```yml
+plugins:
+  - markdown
+  - html
+  - "@typescript-eslint"
+  - react
+  - vue
+```
+
+```bash
+# ESLint checks only `*.js`.
+eslint src docs
+
+# Needs `--ext` option
+eslint src docs --ext .js,.md,.html,.ts,.jsx,.tsx,.vue
+```
+
+However, using plugins which support additional file types is the intention that wants to check those files. The requirement of redundant CLI options is not reasonable.
+
+Per "[Related Discussions](#related-discussions)" section, this is very regularly requested; I want to configure additional target files with `.eslintrc`.
+
+## Detailed Design
+
+This proposal enhances `overrides` property of the config file.
+
+- If a config file in a directory has `overrides` property, ESLint checks the files which are matched by any of override entries (i.e., `files`/`excludedFiles` criteria) additionally in the directory.
+    - If any of `files` value of a override entry ended with `*`, ESLint ignores the override entry to avoid checking all files.
+    - This enhancement affects only the directory path. If people gave glob patterns, ESLint behaves the same as currently.
+    - The ignoring configuration (`.eslintignore`) is prior to this enhancement. If `.eslintignore` contains the additional target files, ESLint just ignores those as same as currently.
+
+The `overrides` property means that people intend to check those files. So this behavior is intuitive.
+
+<table><td>
+💡 <b>Example</b>:
+<pre lang="yml">
+overrides:
+  - files: "*.ts"
+    parser: "plugin:@typescript-eslint/parser"
+  - files: "tests/**/*"
+    env: { mocha: true }
+</pre>
+
+With the above config, `eslint .` command will check `*.ts` files additionally. But the command doesn't check all files inside `tests` directory.
+</td></table>
+
+### Implementation
+
+We can add the check to [lib/cli-engine/file-enumerator.js#L410](https://github.com/eslint/eslint/blob/553795712892c8350b1780e947f65d3c019293a7/lib/cli-engine/file-enumerator.js#L410). It's very easy.
+
+### With file extension processors
+
+If a plugin has file extension processors, [the config array factory yields elements which have `files` property](https://github.com/eslint/eslint/blob/553795712892c8350b1780e947f65d3c019293a7/lib/cli-engine/config-array-factory.js#L870-L902). Therefore this enhancement affects file extension processors.
```

Comment:
It might be a good idea to clarify that this does not affect named processors. (I was initially confused because I thought we were adding implicit behavior when plugins load, but if I'm understanding correctly it only affects the legacy file extension processors, which are already implicit.)

Alternatively, if we want to discourage people from using legacy processors in favor of named processors, maybe we shouldn't implement this for legacy processors. (I'm not sure if legacy processors are deprecated.)

Reply:
Yes, it only affects the legacy file extension processors.

Indeed, #3 says "encourage plugin owners to update their processors to be named instead of extension-named."

Hmm. OK. It's better if implicit behavior is less. I will change this section.